app/javascript/App/components/NewsfeedPage/CoinDrawer.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component, Fragment } from 'react'
+import Swipeable from 'react-swipeable'
+import Animate from 'react-move/Animate'
+import { easeExpOut } from 'd3-ease'
+import CoinList from './CoinList'
+import Icon from '../Icon'
+
+const CoinDrawer = (props) => (
+  <div className="dn-l">
+    <button
+      className="btn btn-blue btn-xs"
+      onClick={() => props.toggleUI('coinDrawer')}
+    >
+      <Icon name="list" className="mr2" />
+      Coin list
+    </button>
+    <Animate
+      show={!!props.currentUI('coinDrawer')}
+      start={{ x: '-100%' }}
+      enter={{
+        x: ['0%'],
+        timing: { duration: 700, ease: easeExpOut }
+      }}
+      leave={{
+        x: ['-100%'],
+        timing: { duration: 200 }
+      }}
+    >
+      {({ x }) => <CoinDrawerList {...props} x={x} />}
+    </Animate>
+  </div>
+)
+
+class CoinDrawerList extends Component {
+  onSwipeLeft = () => {
+    this.props.toggleUI('coinDrawer')
```

Comment:
passing a string to the `toggleUI` method seems less than ideal. it might be interesting to consider passing the constructor name. the `toggleUI` method can handle matching the class to what it should be toggling. then invoking this method would be consistent across the application

Reply:
Don't quite understand this one, maybe you can show me some time